safeguards/parse.js

```
@@ -0,0 +1,34 @@
+'use strict';
+
+const jc = require('json-cycle');
+const YAML = require('js-yaml');
+
+const loadYaml = (contents, options) => {
+  let data;
+  let error;
+  try {
+    data = YAML.load(contents.toString(), options || {});
+  } catch (exception) {
+    error = exception;
+  }
+  return { data, error };
+};
+
+function parse(filePath, contents) {
+  // Auto-parse JSON
+  if (filePath.endsWith('.json')) {
+    return jc.parse(contents);
+  } else if (filePath.endsWith('.yml') || filePath.endsWith('.yaml')) {
+    const options = {
+      filename: filePath,
+    };
+    const result = loadYaml(contents.toString(), options);
+    if (result.error) {
+      throw result.error;
+    }
+    return result.data;
+  }
+  return contents.toString().trim();
```

Shouldn't we crash in that case? It doesn't seem that we support plain string input down the way

---

Sure, we can just throw an error there.

---

Done